SHARE-584 Introduction Task - Simon

simon-costisella opened the pull request:
This is my Introduction Task and I changed the ProjectController.php and project_details.html.twig to add a steal button. I changed the codeView Button to the steal button.

---

### Your checklist for this pull request

Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [ ] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [ ] Choose the proper base branch (_develop_)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no warnings and errors
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [ ] Perform a self-review of the changes
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [ ] Post a message in the _#catroweb_ [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [ ] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Additional Description

`TODO: Add additional information that is not in your commit-message here`

### Tests - additional information

The tests fail because of the missing code view button

`TODO: add additional information about testruns here`

simon-costisella committed:
SHARE-584 Introduction Task Simon

This is my Introduction Task and I changed the ProjectController.php and project_details.html.twig to add a steal button. I changed the codeView Button to the steal button.

dmetzner commented on templates/Project/project_details.html.twig:
```diff
@@ -34,10 +34,10 @@
+        'url': path('projectStealButton', {'id': project.id }),
         'icon': 'extension',
-        'text': "codeview.title"|trans({}, "catroweb")
+        'text': "Steal"
```
Since you have not used the translation system, just a heads up for the future:

`codeview.title` is registered by the developer in the `translations/catroweb.en.yaml` file as 
```
codeview:
    title: Steal
```

All other translation files; E.g `catroweb.de.yml` are translated in its own process (automated by GitHub Actions) using Crowdin -> More details here: https://catrobat.atlassian.net/wiki/spaces/CATWEB/pages/74383377/Crowdin+-+Translation+Management

dmetzner commented on src/Application/Controller/Project/ProjectController.php:
```diff
@@ -212,6 +211,31 @@ public function projectLike(Request $request, string $id): Response
     ]);
   }
 
+    #[Route(path: '/projectStealButton/{id}', name: 'projectStealButton', methods: ['GET'])]
```
At Catroweb we try to write our code using the API approach, to decouple the frontend from the backend. We have multiple different frontends (ours on the share, and also the apps)
Usually, your request (steal) would use our API (https://github.com/Catrobat/Catroweb-API // https://developer.catrobat.org/Catroweb-API/) Of course that is an oerhead for the training example.

Instead of using a GET request, consider using POST which is used to send data to a server to create/update a resource.  Feel free to take a look at all available methods: https://www.w3schools.com/tags/ref_httpmethods.asp

That said it would be nice, to return a status code rather than a redirect in the response, and handle the response using the frontend logic.

dmetzner commented on src/Application/Controller/Project/ProjectController.php:
```diff
@@ -212,6 +211,31 @@ public function projectLike(Request $request, string $id): Response
+        $project->setUser($user);
+        $this->entity_manager->flush();
+
+        $this->addFlash('snackbar', 'Project stolen successfully!');
```
what about the case, when the user already owns the project?

dmetzner commented:
> Found 1 of 903 files that can be fixed in 11.917 seconds, 20.00 MB memory used
Error: Process completed with exit code 8.

With `php composer.phar fix` those code style issues can be resolved automatically

dmetzner commented on templates/Project/project_details.html.twig:
```diff
@@ -63,11 +63,11 @@
     </div>
     <div class="col-6 mb-3 d-lg-none mt-3">
       {% include 'components/redirect_button.twig' with {
```
Usually, a new feature should not replace an existing feature ;)

simon-costisella committed:
SHARE-584 Introduction Task Simon

This is my Introduction Task and I changed the ProjectController.php and project_details.html.twig to add a steal button. I changed the codeView Button to the steal button.

simon-costisella commented:
@schaubes ready for review :)

simon-costisella committed:
fix coding style
